src/main/java/org/scijava/log/LogLevel.java

```diff
@@ -31,66 +31,72 @@
 
 package org.scijava.log;
 
+import java.util.EnumSet;
+
 /**
  * Constants for specifying a logger's level of verbosity.
  * 
  * @author Curtis Rueden
  */
-public final class LogLevel {
+public enum LogLevel {
+
+	NONE(0),
+	ERROR(1), WARN(2), INFO(3), DEBUG(4), TRACE(5);
+
+	public static final EnumSet<LogLevel> VALID_FOR_MESSAGES = EnumSet.range(ERROR, TRACE);
+
+	public static final LogLevel MAX_VALUE = TRACE;
+
+	public static final LogLevel MIN_VALUE = NONE;
+
+	private final int intLevel;
 
-	private LogLevel() {
-		// prevent instantiation of utility class
+	LogLevel(int intLevel) {
+		this.intLevel = intLevel;
 	}
 
-	public static final int NONE = 0;
-	public static final int ERROR = 1;
-	public static final int WARN = 2;
-	public static final int INFO = 3;
-	public static final int DEBUG = 4;
-	public static final int TRACE = 5;
-
-	public static String prefix(final int level) {
-		switch (level) {
-			case ERROR:
-				return "[ERROR] ";
-			case WARN:
-				return "[WARNING] ";
-			case INFO:
-				return "[INFO] ";
-			case DEBUG:
-				return "[DEBUG] ";
-			case TRACE:
-				return "[TRACE] ";
-			default:
-				return "[LEVEL:" + level + "] ";
-		}
+	public int intLevel() {
+		return intLevel;
 	}
 
-	/**
-	 * Extracts the log level value from a string.
-	 * 
-	 * @return The log level, or -1 if the level cannot be parsed.
-	 */
-	public static int value(final String s) {
-		if (s == null) return -1;
-
-		// check whether it's a string label (e.g., "debug")
-		final String log = s.trim().toLowerCase();
-		if (log.startsWith("n")) return LogLevel.NONE;
-		if (log.startsWith("e")) return LogLevel.ERROR;
-		if (log.startsWith("w")) return LogLevel.WARN;
-		if (log.startsWith("i")) return LogLevel.INFO;
-		if (log.startsWith("d")) return LogLevel.DEBUG;
-		if (log.startsWith("t")) return LogLevel.TRACE;
-
-		// check whether it's a numerical value (e.g., 5)
-		try {
-			return Integer.parseInt(log);
-		}
-		catch (final NumberFormatException exc) {
-			// nope!
+	public String prefix() {
+		return "[" + toString() + "] ";
+	}
+
+	public static LogLevel valueOf(int intLevel) {
+		switch (intLevel) {
+			case 0: return NONE;
+			case 1: return ERROR;
+			case 2: return WARN;
+			case 3: return INFO;
+			case 4: return DEBUG;
+			case 5: return TRACE;
 		}
-		return -1;
+		throw new IllegalArgumentException();
+	}
+
+	public boolean isHigherThan(LogLevel level) {
```

Comment:
Actually, there is one single piece of metadata: the label. E.g., 0=NONE, 3=INFO. This would be nice if it were extensible... and actually, it's easy to do so by following the extensible enum pattern. Take a look at `net.imagej.axis.AxisType` and `net.imagej.axis.Axes`. Would this sort of pattern satisfy you here, @maarzt? It is significantly more complex, but would allow custom log levels to name themselves. Otherwise, we could achieve the same thing by adding a method to `LogService` which lets people register new labels with custom log levels. This seems overly fancy and potentially error-prone to me, since we need to implement lots of methods to support it. E.g.: what if two `LogLevel`s exist with the same `int` level but different labels? Easier to simply stick with `int`.